tests/fastify-plugins/maps.js

```diff
@@ -123,3 +204,102 @@ function setup(t) {
 
   return server
 }
+
+function setupFetchMock() {
+  const mockAgent = new MockAgent({
+    keepAliveMaxTimeout: 10,
+    keepAliveTimeout: 10,
+  })
+
+  mockAgent.disableNetConnect()
+
+  setGlobalDispatcher(mockAgent)
```

Comment:
Do we ever undo this global dispatcher?

Reply:
in its current implementation, no. was wondering if i needed to and had a previous implementation where i basically called this setup for each test so that I could access the teardown. maybe that's the safe way to go?

i think part of it depends on how the test runner works. does brittle run all tests in the same process? if so, will probably need some kind of cleanup

Reply:
the hacky approach is to add a final `test` that does a cleanup e.g.


test('cleanup fetch mock', (t) => {
  t.teardown(() => {
    cleanupFetchMock()
  })
})

Reply:
Alternatively, you could put everything in a `test` block and then have subtests. Something vaguely like:

```js
test('outer test', async (t) => {
  setupFetchMock()
  t.teardown(cleanupFetchMock)

  t.test('subtest 1', async (st) => { /* ... */ })

  t.test('subtest 2', async (st) => { /* ... */ })

  // ...
})
```

Reply:
i'll leave it as is for now if that's okay then!